### PR TITLE
Chore: remove lib/rocketchat step 2

### PR DESCRIPTION
**Pull request by GleidsonDaniel**
## Proposed changes

Removes Rocketchat.method calls and calls them directly.

## How to test or reproduce

- Share extension
- Push notification
- All `+` features from inside the room
- Jitsi
- Connection
- Deep linking

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/views/CreateDiscussionView/index.tsx [hunk 1]
```diff
@@ -67,7 +67,7 @@ class CreateChannelView extends React.Component<ICreateChannelViewProps, any> {
 					}
 					const item = {
 						rid,
-						name: RocketChat.getRoomTitle(result),
+						name: getRoomTitle(result),
```
Comment (diegolmello) [inline app/views/CreateDiscussionView/index.tsx:70]: I couldn't get the reason, but we can't create discussions on this PR.

**Review by diegolmello — CHANGES_REQUESTED** [review]
At this point we agreed to switch to `Services.service`.
In the future, we're going to migrate to a custom hook instead.

File: app/lib/methods/triggerActions.ts [hunk 2]
```diff
@@ -5,21 +5,20 @@ import {
...
 import Navigation from '../navigation/appNavigation';
 import { triggerAction } from './actions';
 
-export function triggerBlockAction(this: TRocketChat, options: ITriggerBlockAction) {
-	return triggerAction.call(this, { type: ActionTypes.ACTION, ...options });
-}
-
-export async function triggerSubmitView(this: TRocketChat, { viewId, ...options }: ITriggerSubmitView) {
-	const result = await triggerAction.call(this, { type: ActionTypes.SUBMIT, viewId, ...options });
+export async function triggerSubmitView({ viewId, ...options }: ITriggerSubmitView) {
```
[the remark this answers is not in the document]
Comment (GleidsonDaniel) [reply app/lib/methods/triggerActions.ts:11]: In the end they call triggerAction and this function doesn't use this.

File: app/lib/methods/helpers.ts [hunk 3]
```diff
@@ -89,63 +85,6 @@ export function hasRole(role): boolean {
...
-	}
-	return JSON.parse(allowAnalyticsEvents);
-}
-
-// TODO: remove this
-export async function subscribeRooms(this: any) {
-	if (!this.roomsSub) {
-		try {
-			// TODO: We need to change this naming. Maybe move this logic to the SDK?
-			this.roomsSub = await subscribeRoomsTmp.call(this);
```
[the remark this answers is not in the document]
Comment (GleidsonDaniel) [reply app/lib/methods/helpers.ts:114]: This function has been moved to another file.

File: app/lib/methods/index.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
+export * from './actions';
+export * from './callJitsi';
+export * from './canOpenRoom';
+export * from './clearCache';
+export * from './enterpriseModules';
+export * from './getCustomEmojis';
+export * from './getPermalinks';
+export * from './getPermissions';
+export * from './getRoles';
+// export * from './getRoom'; only used inside methods folder
```
[the remark this answers is not in the document]
Comment (GleidsonDaniel) [reply app/lib/methods/index.ts:10]: More or less, their signature is different, one returns throw and another null, the code will change to unite them, I think it's better to do this in another task.

File: app/lib/methods/getSingleMessage.ts [hunk 5]
```diff
@@ -1,10 +1,10 @@
-import RocketChat from '../rocketchat';
 import { IMessage } from '../../definitions';
+import { getSingleMessage } from '../services';
 
-const getSingleMessage = (messageId: string): Promise<IMessage> =>
+const getMessage = (messageId: string): Promise<IMessage> =>
```
[the remark this answers is not in the document]
Comment (GleidsonDaniel) [reply app/lib/methods/getSingleMessage.ts]: I think we should create a task to clean up redundant stuff and the like.

File: app/lib/methods/getRoomInfo.ts [hunk 6]
```diff
@@ -9,7 +9,7 @@ export interface IRoomInfoResult {
 	t: IServerSubscription['t'];
 }
 
-const getRoomInfo = async (rid: string): Promise<IRoomInfoResult | null> => {
+const getInfo = async (rid: string): Promise<IRoomInfoResult | null> => {
```
[the remark this answers is not in the document]
Comment (GleidsonDaniel) [reply app/lib/methods/getRoomInfo.ts]: Just changed the file.

**Review by diegolmello — APPROVED** [review]

**Pull request merged**